components/merkledb/src/macros.rs

```diff
@@ -152,3 +152,47 @@ macro_rules! impl_serde_hex_for_binary_value {
         }
     };
 }
+
+/// Implements `BinaryValue` traint for tuples cross single member generic tuple struct.
+/// If expression begins with for keyword it creates implementation for existend structure
+/// supposing it meets requirements, mentioned above. Otherwise it creates new structure
+/// according to supplied meta attributes, visibility, name and type constraint for generic type.
+/// Target tuple types must be listed in curly braces.
```

Comment:
An example would be much better than a documentation. Also, consider the following fixes in grammar:

```
If an expression begins with `for` keyword, it creates an implementation for existent structure 
that meets the requirements mentioned above. Otherwise it creates a new structure 
according to supplied meta attributes, visibility, name and type constraints for a generic type.
```